pyrdp/mitm/DeviceRedirectionMITM.py

```diff
@@ -4,146 +4,54 @@
 # Licensed under the GPLv3 or later.
 #
 
-import datetime
 import hashlib
 import json
 from logging import LoggerAdapter
 from pathlib import Path
-from typing import BinaryIO, Dict
+from typing import Dict, Optional, Union
 
-import names
-
-from pyrdp.core import decodeUTF16LE
-from pyrdp.enum import CreateOption, DeviceType, FileAccess, IOOperationSeverity
+from pyrdp.core import FileProxy, ObservedBy, Observer, Subject
+from pyrdp.enum import CreateOption, DeviceType, DirectoryAccessMask, FileAccessMask, FileAttributes, \
+    FileCreateDisposition, FileCreateOptions, FileShareAccess, FileSystemInformationClass, IOOperationSeverity, \
+    MajorFunction, MinorFunction
 from pyrdp.layer import DeviceRedirectionLayer
 from pyrdp.mitm.config import MITMConfig
-from pyrdp.parser import DeviceRedirectionParser
-from pyrdp.pdu import DeviceCloseRequestPDU, DeviceCreateRequestPDU, \
-    DeviceIORequestPDU, DeviceIOResponsePDU, DeviceListAnnounceRequest, DeviceReadRequestPDU, DeviceRedirectionPDU
+from pyrdp.mitm.FileMapping import FileMapping, FileMappingDecoder, FileMappingEncoder
+from pyrdp.pdu import DeviceAnnounce, DeviceCloseRequestPDU, DeviceCloseResponsePDU, DeviceCreateRequestPDU, \
+    DeviceCreateResponsePDU, DeviceDirectoryControlResponsePDU, DeviceIORequestPDU, DeviceIOResponsePDU, \
+    DeviceListAnnounceRequest, DeviceQueryDirectoryRequestPDU, DeviceQueryDirectoryResponsePDU, DeviceReadRequestPDU, \
+    DeviceReadResponsePDU, DeviceRedirectionPDU
 
 
-class FileMapping:
-    """
-    Class containing information for a file intercepted by the DeviceRedirectionMITM.
-    """
+class DeviceRedirectionMITMObserver(Observer):
```

Comment:
It's a base class for the observer of the component. I don't want to mark the methods as abstract because I don't think it's necessary to force implementors to handle ALL messages (e.g: maybe a custom implementor is only interested in directory listing and doesn't care about file requests).